### ISSUE TITLE
Incorrect results for `floatDiffInDays()`.

### ISSUE THREAD

**Issue by amcsi** [briannesbitt/Carbon#2255]
Hello,

I encountered an issue with the following code:
```php
>>> (new \Carbon\CarbonImmutable('2020-12-17 00:00:00.000001'))->floatDiffInDays(new \Carbon\CarbonImmutable('2020-12-16 00:00:00.000000'))
=> 1.0000000000116
// ^ good

>>> (new \Carbon\CarbonImmutable('2020-12-17 00:00:00.000001'))->floatDiffInDays(new \Carbon\CarbonImmutable('2020-12-16 00:00:00.000001'))
=> 1.0
// ^ good

>>> (new \Carbon\CarbonImmutable('2020-12-17 00:00:00.000001'))->floatDiffInDays(new \Carbon\CarbonImmutable('2020-12-16 00:00:00.000002'))
=> 1.9999999999884
// ^ bad; should be 0.999999..., because the difference is one day minus a microsecond.
```

Carbon version: **2.42.0**

PHP version: **7.4 and 8.0**

I expected to get:

```
0.999999...
```

But I actually get:

```
1.9999999999884
```

Thanks!

**Comment by kylekatarnls** [comment]
Thanks for the report.

For the record, note that it's fixed on `3.x-dev` in progress next major version. So maybe the new logic could be downgraded in 2.x.

**Comment by kylekatarnls** [comment]
It will be fixed in 2.43.0, but due to a bug in PHP, it will only work properly with PHP >= 7.2.
